src/outputs/ascent.cpp

```diff
@@ -0,0 +1,203 @@
+//========================================================================================
+// Parthenon performance portable AMR framework
+// Copyright(C) 2023 The Parthenon collaboration
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+// (C) (or copyright) 2023. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+//! \file ascent.cpp
+//  \brief Ascent situ visualization and analysis interop
+
+#include <cstdio>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+
+// Parthenon headers
+#include "coordinates/coordinates.hpp"
+#include "defs.hpp"
+#include "globals.hpp"
+#include "mesh/mesh.hpp"
+#include "outputs/outputs.hpp"
+#include "utils/error_checking.hpp"
+
+// Ascent headers
+#ifdef PARTHENON_ENABLE_ASCENT
+#include "ascent.hpp"
+#include "conduit_blueprint.hpp"
+#include "conduit_relay_io.hpp"
+#include "conduit_relay_io_blueprint.hpp"
+#endif // ifdef PARTHENON_ENABLE_ASCENT
+
+namespace parthenon {
+
+//----------------------------------------------------------------------------------------
+//! \fn void AscentOutput:::WriteOutputFile(Mesh *pm)
+//  \brief  Expose mesh and all Cell variables for processing with Ascent
+void AscentOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm,
+                                   const SignalHandler::OutputSignal signal) {
+#ifndef PARTHENON_ENABLE_ASCENT
+  if (Globals::my_rank == 0) {
+    PARTHENON_WARN("Ascent output requested by input file, but Ascent support not "
+                   "compiled in. Skipping this output type.");
+  }
```

Comment:
Do we want this printed every ascent output? I guess that's not unreasonable... But I think I might prefer a quieter error message.

Reply:
On second thought, how about a loud fail?
I'm coming from the point of view that if you expect an output it should be there,not that you run a sim and wonder where the files are after burning X hours of computing time.

Reply:
I find loud failure for outputs in particular annoying since I often copy around input decks and it's irritating to have to comment out blocks to get the code to run. But I don't feel too strongly either way.

Reply:
I see your point.
I left it as it for now (as it doesn't add  new logic/idefs at other places).

Reply:
:+1: yeah that's fine